components/command_palette/command_palette_modal/command_palette_modal.tsx

```diff
@@ -53,41 +53,31 @@ const CommandPaletteModal = ({onExited, selectedEntities}: Props) => {
     const currentTeamId = useSelector(getCurrentTeamId);
     const teams = useSelector(getTeams);
 
-    const searchHandler = useSelector((state: GlobalState) => state.plugins.searchHandlers.focalboard);
-    const recentHandler = useSelector((state: GlobalState) => {
-        return state.plugins.recentlyViewedHandlers.focalboard;
-    });
+    const recentlyViewedHandlers = useSelector((state: GlobalState) => state.plugins.recentlyViewedHandlers);
+    const searchHandlers = useSelector((state: GlobalState) => state.plugins.searchHandlers);
+
     const pluginsList = useSelector((state: GlobalState) => state.plugins.plugins);
-    const boards = pluginsList.focalboard;
-    const playbooks = pluginsList.playbooks;
     const recentChannels = useSelector(getChannelsInAllTeams);
     const channelsTransformedItems = channelToCommandPaletteItemTransformer(recentChannels, teams);
-    const boardsTransformedItems: CommandPaletteItem[] = [];
-    const recentBoards = useCallback(async () => {
-        // const recentData: any = await recentHandler();
-        // if (recentData) {
-        //     boardsTransformedItems = boardToCommandPaletteItemTransformer(recentData, teams);
-        //     setTransformedItems([...boardsTransformedItems, ...channelsTransformedItems]);
-        //     setLoading(false);
-        // }
-        setTransformedItems([...channelsTransformedItems]);
+
+    // recentlyViewed synchronously fetches recently viewed items from all registered handlers.
+    const recentlyViewed = useCallback(async () => {
+        let items = Object.values(recentlyViewedHandlers).map((f) => f());
+        setTransformedItems([...defaultsTransformer(items.flat(), teams), ...channelsTransformedItems]);
```

Comment:
Note that I'm proposing we make it the plugin's responsibility to provide the "right fields", i.e. fit the interface, with `defaultsTransformer` just handling the team name and tidying anything else up that's required.

This should absolve the webapp from having to hard-code any mapping knowledge.

Reply:
Yeah make sense, we export interface/type via our type package; are we using the type package in playbooks? @harshilsharma63 same question for boards